### PR TITLE
Preprocessing

### PR DESCRIPTION
**Pull request by esarbanis**
Utilized `svelte.preprocess` introduced to svelte in v1.44.0

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -31,8 +32,8 @@ module.exports = function(source, map) {
 
 	if (!options.name) options.name = capitalize(sanitize(options.filename));
 
-	try {
-		let { code, map, css, cssMap } = compile(source, options);
+	preprocess(source, options).then(processed => {
```
Comment (nikku) [inline index.js]: Use tabs instead of spaces.

File: test/loader.spec.js [hunk 2]
```diff
@@ -253,6 +261,79 @@ describe('loader', () => {
...
+				};
+
+				loader.call(
+					{
+						cacheable: cacheableSpy,
+						async: () => callbackSpy,
+						resourcePath: 'test/fixtures/style-valid.html',
+						options,
+					},
+            fileContents,
```
Comment (nikku) [inline test/loader.spec.js]: Fix formating; use tabs instead of spaces.

File: index.js [hunk 3]
```diff
@@ -44,10 +45,10 @@ module.exports = function(source, map) {
 			utimesSync(tmpobj.name, stats.atimeMs - 9999, stats.mtimeMs - 9999);
 		}
 
-		this.callback(null, code, map);
```
Comment (nikku) [inline index.js]: Formating: use tabs instead of spaces.

File: index.js (hunk 1 above)
Comment (esarbanis) [reply index.js]: Will do, thanks!

**Comment by esarbanis** [comment]
The weird thing is that I do have adapted the test cases to be async and they all pass in my local machine. 

It could be some sort of permission limitation in travis, cause we write in a temp file, although  `/tmp` should not need special permissions ... I really don't know what's going on with that test ...

**Comment by esarbanis** [comment]
Found the issue in tests, in node 8.0.0 the `Stats` object [does not contain an `atimeMs` or `mtimeMs` fields](https://github.com/nodejs/node/blob/v8.0.0/doc/api/fs.md#class-fsstats), so bumped node's version up to a more recent 8.x version

**Comment by esarbanis** [comment]
Hm, now that I think of it we should better make the loader compatible with earlier versions of node.

I updated the change.

**Comment by nikku** [comment]
> Hm, now that I think of it we should better make the loader compatible with earlier versions of node.

Please ensure that we test this on travis, too. Just add the additional node versions you'd like to test to `.travis.yml`.

In my opinion testing `stable` and `lts` would be sufficient (cf. [Travis docs](https://docs.travis-ci.com/user/languages/javascript-with-nodejs/).

**Comment by esarbanis** [comment]
On it!

**Comment by esarbanis** [comment]
According to [this](https://github.com/nodejs/Release) we should cover 6.x, 8.x and latest 9. The problem with the previous error was that from 8.0 the api changed, within the 8.x range. Anyway, I suggest the following:
```
node_js:
- '9'
- '8'
- '8.0'
- '6'
```

**Comment by esarbanis** [comment]
OK, putting `8.0` is just dump as we cover the api changes from `6.x` to `9`! I removed it.

**Comment by Rich-Harris** [comment]
Thank you! Released as 2.3.0 🎉

**Pull request merged**